adapters/github.go

```diff
@@ -146,11 +146,8 @@ func (j *jwtAuthTransportWrapper) RoundTrip(req *http.Request) (*http.Response,
 }
 
 func NewGithubClient(config GitHubClientConfig) (*GithubClient, error) {
-	if config.HTTPClient == nil {
-		config.HTTPClient = http.DefaultClient
-	}
-
 	// We need to modify the transport for authentication if credentials are provided
+	// If the config.HTTPClient is nil, we pass it as it is
 	httpClient := config.HTTPClient
```

Comment:
@Sahilb315 Can you review the entire `github.go` and see if we should remove all usage of `http.DefaultClient` and keep it as nil?